cmd/wasm-client/main.go

```diff
@@ -103,24 +84,55 @@ func main() {
 		exit(err)
 	}
 
+	wc := newWasmClient(&cfg)
+
+	// Setup JS callbacks.
+	js.Global().Set("wasmClientIsReady", js.FuncOf(wc.IsReady))
+	js.Global().Set("wasmClientConnectServer", js.FuncOf(wc.ConnectServer))
+	js.Global().Set("wasmClientIsConnected", js.FuncOf(wc.IsConnected))
+	js.Global().Set("wasmClientDisconnect", js.FuncOf(wc.Disconnect))
+	js.Global().Set("wasmClientInvokeRPC", js.FuncOf(wc.InvokeRPC))
+
+	for _, registration := range registrations {
+		registration(wc.registry)
+	}
+
 	log.Debugf("WASM client ready for connecting")
 
 	select {
 	case <-shutdownInterceptor.ShutdownChannel():
 		log.Debugf("Shutting down WASM client")
-		_ = wasmClientDisconnect(js.ValueOf(nil), nil)
+		_ = wc.Disconnect(js.ValueOf(nil), nil)
 		log.Debugf("Shutdown of WASM client complete")
 	}
 }
 
-func wasmClientIsReady(_ js.Value, _ []js.Value) interface{} {
+type wasmClient struct {
+	cfg *config
+
+	lndConn *grpc.ClientConn
+
+	registry map[string]func(context.Context, *grpc.ClientConn,
+		string, func(string, error))
+}
+
+func newWasmClient(cfg *config) *wasmClient {
+	return &wasmClient{
+		cfg:     cfg,
+		lndConn: nil,
+		registry: make(map[string]func(context.Context,
```

Comment:
nit: can use `stubPackageRegistration` here as an abbreviation for the function type.

Reply:
they are slightly different though. `stubPackageRegistration`'s type is a _func_ that calls this type.

i can add a new type for this though it it would improve readability?

Reply:
Ah, I didn't spot the difference. Nah, let's leave it then.